Automattic/wp-calypso
# [GBonWPCOM e2e tests i0] Add assertion for blocks in the frontend (published) page for edge/non-edge, add @parallel tag

## Changes proposed in this Pull Request

This is a follow up for https://github.com/Automattic/wp-calypso/pull/45048. Tracking issue: https://github.com/Automattic/wp-calypso/issues/45209.

- The `@parallel` tag (one per block/edge+non edge site test)has been added so that the test is picked up by magellan/CI and will run faster (< 6min timeout limit);
- Initial groundwork that checks for the blocks that should be rendered; this means adding a selector to the block classes for the blocks that already render on the frontend.
- Some blocks require additional setup before appear on the frontend and are not included as part of this changeset (see other related PRs).
- Fixes an issue that was preventing screenshots from being created on the CI VM.


This was extracted from the PR here: https://github.com/Automattic/wp-calypso/pull/45284. We decided to split it into several PRs, instead. The amount of files changed is the same (due to the static property being added to the block classes with the frontend selector) but the actual changeset is simpler, as the logic for filling/setting-up each block has been extracted into other PRs.

⚠️ The goal is to review and merge this **first** so that the other related PR's (see below) diffs become simpler and are easier to review and merge after this one ⚠️ 

## Why

The[ initial task](https://github.com/Automattic/wp-calypso/pull/45048) added the framework/PoC for automating our manual testing process. It only configured two blocks, though. The goal of this PR is to lay the groundwork to later fill in all of the popular blocks being tested as part of the upgrade E2E test, and verify them in the published page too (edge/non-edge and across the most popular themes).

## Testing instructions

1. Setup your env to run Calypso E2E tests (search for "Automated end-to-end Testing") in the FG;
1. Run with `./node_modules/.bin/mocha specs/wp-calypso-gutenberg-upgrade-spec.js`*[0]
1. Wait for the tests to finish. It might take a good while since it tests everything for each test site
1. Make sure it didn't err, if it did, it might not be an actual failure, try running that specific step again. If it continues failing, let us know
1. Add the `Needs e2e Testing Gutenberg Edge` to this PR (already added, if necessary, remove and add again);
1. Check this branch's tests in CI: [Desktop](https://circleci.com/workflow-run/376ad2ba-2c34-4883-a731-5619c8fa5d0f), [Mobile](https://circleci.com/workflow-run/1fdee30c-7da0-4398-be7d-f17b6be5ff63).

*[0] _Make sure to cd into `test/e2e` first._

### Related PRs

Review and test the following PRs after this PR has been merged:

* https://github.com/Automattic/wp-calypso/pull/45795
* https://github.com/Automattic/wp-calypso/pull/45796
* https://github.com/Automattic/wp-calypso/pull/45797
*  https://github.com/Automattic/wp-calypso/pull/45798
* https://github.com/Automattic/wp-calypso/pull/45799
